src/net_processing.cpp

```diff
@@ -4135,6 +4135,11 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
     if (msg_type == NetMsgType::GETBLOCKTXN) {
         BlockTransactionsRequest req;
         vRecv >> req;
+        // Verify differential encoding invariant: indexes must be strictly increasing
+        // DifferenceFormatter should guarantee this property during deserialization
+        for (size_t i = 1; i < req.indexes.size(); ++i) {
+            assert(req.indexes[i] > req.indexes[i-1]);
```

Comment:
```suggestion
            Assume(req.indexes[i] > req.indexes[i-1]);
```

Just so the assertion isn't active in release builds

Reply:
yes. thanks